### PR TITLE
Fix for #162 - Deleting Specs

### LINKED ISSUE #162: Spec deletion not working in practice
See: https://github.com/CocoaPods/CocoaPods/issues/4743

### PR DESCRIPTION
**Pull request by orta**
Fixes #162.

I think this is what is needed to get spec deletion working. 
Want to get back to CocoaPods.app though.

### CODE REVIEW

File: app/models/github.rb [hunk 1]
```diff
@@ -33,21 +33,33 @@ def create_new_commit(destination_path, data, message, author_name, author_email
         end
       end
 
-      def delete_file_at_path(destination_path, message, author_name, author_email)
+      def delete_file_at_path(destination_path, message, sha, author_name, author_email)
         CommitResponse.new do
           delete(File.join('contents', URI.escape(destination_path)),
-                 :message   => message,
-                 :branch    => BRANCH,
+                 :Message   => message,
```
Comment (orta) [inline app/models/github.rb]: heh, oh emacs

File: app/models/github.rb [hunk 2]
```diff
@@ -33,21 +33,33 @@ def create_new_commit(destination_path, data, message, author_name, author_email
...
+                 :sha => sha,
                  :author    => { :name => author_name,        :email => author_email },
                  :committer => { :name => ENV['GH_USERNAME'], :email => ENV['GH_EMAIL'] }
           )
         end
       end
 
+      def sha_for_path(path)
+        CommitResponse.new do
+          get(File.join('contents', URI.escape(destination_path), ".json")
```
Comment (orta) [inline app/models/github.rb]: `path` not `destination_path`

File: app/models/github.rb (hunk 1 above)
Comment (segiddins) [inline app/models/github.rb]: Why capitalize this key?

**Comment by segiddins** [comment]
CI is failing. I'll review this later today when I can test out the github API.

File: app/models/github.rb (hunk 1 above)
Comment (orta) [inline app/models/github.rb]: one of the really common key combos for file navigation is right next to the one that capitalises

File: app/models/github.rb [hunk 3]
```diff
@@ -33,21 +33,33 @@ def create_new_commit(destination_path, data, message, author_name, author_email
...
-                 :branch    => BRANCH,
+                 :Message   => message,
+                 :sha => sha,
                  :author    => { :name => author_name,        :email => author_email },
                  :committer => { :name => ENV['GH_USERNAME'], :email => ENV['GH_EMAIL'] }
           )
         end
       end
 
+      def sha_for_path(path)
```
Comment (segiddins) [inline app/models/github.rb]: docs

File: app/models/github.rb [hunk 4]
```diff
@@ -98,6 +110,10 @@ def body
           @response.body
         end
 
+        def header(name)
```
Comment (segiddins) [inline app/models/github.rb]: docs

File: app/models/github.rb [hunk 5]
```diff
@@ -33,21 +33,33 @@ def create_new_commit(destination_path, data, message, author_name, author_email
...
       end
 
-      def delete_file_at_path(destination_path, message, author_name, author_email)
+      def delete_file_at_path(destination_path, message, sha, author_name, author_email)
         CommitResponse.new do
           delete(File.join('contents', URI.escape(destination_path)),
-                 :message   => message,
-                 :branch    => BRANCH,
+                 :Message   => message,
+                 :sha => sha,
```
Comment (segiddins) [inline app/models/github.rb]: align the `=>`

File: app/models/github.rb (hunk 2 above)
Comment (segiddins) [inline app/models/github.rb]: missing a closing `)`

**Comment by orta** [comment]
I've cleaned up this PR, should be everything but specs

File: app/models/push_job.rb [hunk 6]
```diff
@@ -1,8 +1,10 @@
+# coding: utf-8
```
Comment (segiddins) [inline app/models/push_job.rb]: unnecessary

File: app/models/push_job.rb [hunk 7]
```diff
@@ -42,15 +44,16 @@ def perform_action
...
+          contents_response = self.class.github.sha_for_path(pod_version.destination_path).body
+          sha = JSON.parse(contents_response)['sha']
           self.class.github.delete_file_at_path(
             pod_version.destination_path,
             commit_message,
+            sha,
             committer.name,
-            committer.email
-          )
+            committer.email)
```
Comment (segiddins) [inline app/models/push_job.rb]: these `)` should be on the next line for consistency

**Comment by orta** [comment]
OK cleaned up and added docs, gonna try look at comments

File: app/models/push_job.rb [hunk 8]
```diff
@@ -42,12 +43,14 @@ def perform_action
             specification_data,
             commit_message,
             committer.name,
-            committer.email
-          )
+            committer.email)
```
Comment (segiddins) [inline app/models/push_job.rb]: `)` should be on the next line

File: app/models/github.rb [hunk 9]
```diff
@@ -33,21 +33,39 @@ def create_new_commit(destination_path, data, message, author_name, author_email
...
+                 :sha       => sha,
                  :author    => { :name => author_name,        :email => author_email },
                  :committer => { :name => ENV['GH_USERNAME'], :email => ENV['GH_EMAIL'] }
           )
         end
       end
 
+      # @return [CommitResponse] A encapsulated response object that gets the SHA associated with a file at a path
+      #
+      def sha_for_path(path)
```
Comment (segiddins) [inline app/models/github.rb]: this should be `file_at_path` since it doesn't return the sha

File: app/models/push_job.rb (hunk 8 above)
Comment (orta) [inline app/models/push_job.rb]: likely `rubocop -a`

Comment (orta) [inline app/models/push_job.rb]: or not, was me

**Comment by orta** [comment]
Alright, I've got myself covered, and it caught a bug :+1:

**Comment by orta** [comment]
OK, this should be good :+1:

**Comment by segiddins** [comment]
It's no longer a WIP?

**Comment by orta** [comment]
Was just doing it :dancers:

**Comment by orta** [comment]
https://github.com/CocoaPods/trunk.cocoapods.org/pull/166 continues on from this

**Pull request merged**